### ISSUE TITLE
Confirm transaction book doesn't appear in 1.14

### ISSUE THREAD

**Issue by MadMick96** [MLG-Fortress/PrettySimpleShop#12]
Hi,

This is not a gamebreaking issue by any means but slightly annoying. 
In order to buy from a chest twice in a row, or from two different chests, I need to say something in chat in between buying. 

So if I:

Click on chest selling "Leather", and it asks me how many I want to buy and I type 2, nothing happens
I then type "x" in chat, and click again (on this chest or another shop chest) and type 2 after it asks me for quantity, it works. I'm playing Spigot 1.14.1, EssentialsX economy. There are no error messages in console.

Basically: always need to type something in chat in order to buy anything for the first time or after buying something else.

**Comment by RoboMWM** [comment]
You need to click the same chest twice for the chat prompt to function. This is so you're "double clicking" to buy vs. just viewing info. It also prevents spamming of the "how many would you like to buy" prompt.

It was this way since before you would use the /buy command or click the listing in chat. Now that the trade confirmation feature is in place, I could consider having the prompt on just the first click... But I don't want the prompt to spam when a person is just viewing what's for sale. So I'm not sure what the best way to go about this would be; it seems most players on my server easily figure out how to buy things (since most new players are used to punching a sign/chest to buy stuff.)

**Comment by MadMick96** [comment]
It seems that even though I click twice, I cannot immediately buy something, but rather I have to try to buy something (and fail) and then the next attempt will work, I'll try to get a gif for you. I am indeed double clicking and getting the prompt "how many would you like to buy?" but the chat prompt still doesn't function ( also doesn't say the number in chat but the prompt just literally does nothing)

**Comment by MadMick96** [comment]
https://gyazo.com/8a773d0cd256b077414d049e5fbd6de8

Here is a gif explaining approximately what I mean.

**Comment by RoboMWM** [comment]
Oh, what client version is that? If it's 1.8 or lower then you should disable the transactionConfirmation feature since the plugin message changed over MC version updates. A book should've appeared showing you a preview of the transaction, but in your case since it looks like your client is old, it won't recognize the plugin message. Running the transaction again will confirm it hence why it appears to works the second time.

**Comment by MadMick96** [comment]
It's 1.14.2

**Comment by RoboMWM** [comment]
Hmm, maybe the plugin message changed in 1.14 then.

**Comment by RoboMWM** [comment]
Just to confirm, what's your config?

**Comment by ichbinashh** [comment]
I'd just like to add that I have just tried installing on 1.14.2 and get this exact same issue.

Clicking once displays info from the shop, clicking twice prompts to buy. Enter a number, nothing happens. Click it again and the prompt reoccurs, enter a number this time and it works. I am also not getting a confirmation before the transaction completes.

Freshly installed, so default config is used.

**Comment by RoboMWM** [comment]
I'm assuming if confirmations are disabled it will work?

**Comment by ichbinashh** [comment]
I haven't tried that, but I will give it a go later and report back

**Comment by ichbinashh** [comment]
That is exactly the case, works perfectly with them disabled.

**Comment by RoboMWM** [comment]
> [22:24:21] RoboMWM: did the book open protocol disappear or am I just not looking hard enough in wiki.vg
> [22:24:41] RoboMWM: *plugin message
> [22:26:37] Brokkoli: its now a packet
> [22:27:11] Brokkoli: and spigot 1.14 has an api for it, so you do not need direct packet handling any more
> [22:27:37] Brokkoli: https://wiki.vg/Pre-release_protocol#Open_Book
> [22:28:04] Brokkoli: https://hub.spigotmc.org/javadocs/spigot/org/bukkit/entity/Player.html#openBook-org.bukkit.inventory.ItemStack-
